_posts/2020-12-15-why-developer-should-know-owasp.md

```diff
@@ -0,0 +1,71 @@
+
+Of course, there are software projects where the baseline of the information security must be higher, and achieving that level costs more money. However, I like to think that securely implementing basic things does not increase costs. If you decide to do, for example, SQL queries in a way where SQL injection is impossible from the beginning, it takes as much time as making them less securely. The costs increase only if you first make them less securely, and then in a more secure way afterward.
+
+![Bobby Tables we call him](/img/why-owasp/exploits_of_a_mom.png)
+
+*Bobby Tables we call him, a classic from [https://xkcd.com/327/](https://xkcd.com/327/)*
+
+In a software project, the possible risks must be examined carefully, and sufficient security level needs to be decided. If the information system is going to be used in a restricted network by a certain number of users, you might not need the same amount of security as in open banking systems. However, it is a decision which needs to be done knowingly. 
+
+# Where the information security comes from?
+
+The information security in a project comes from people. Depending on the project there might be some security specialists but all the people involved in the project need to aware of security issues.
+
+One security aspect is the practises used in the project. Do people behave securely? When working on a project, you can't tell everyone about it, at least not at a detailed level. You must not keep your passwords on post-its, you should use always good passwords, perhaps utilize a password manager, and enable multi-factor authentication whenever it's possible. Do not push sensitive data, such as database passwords, to version control. Use a VPN instead of unprotected WLANs. Recognize if the risk is bearable before pasting project-specific things into third-party services, like Pastebin, XML validator or some language translator. After all that the other aspect is being aware of different security threats which exist in a digital environment. 
+
+![Passwords on post-its](/img/why-owasp/passwords.jpg)
+
+The developers are basically problem solvers. We need to solve a problem which is given to us. Sometimes the problem is easy and simple to solve. However, just solving the problem might not be enough. For example, a testing specialist might find a bug from the implementation in seconds. They have a different point of view and typically try all the nasty corner cases. Does this work with empty input? Or with negative numbers? If the developer is experienced, they have been already taken care of all those corner cases. Also, hackers have a different point of view. They are used to finding vulnerabilities from the software. To be able to secure the software, also the developers must be aware of the threats. 

```

Comment:
Change to: "If the developers are experienced, they have already"...

Reply:
good point